go/service/chat_local.go

```diff
@@ -293,33 +371,35 @@ func (h *chatLocalHandler) getConversationInfo(ctx context.Context, conversation
 		MessageIDs: messageIDs,
 	})
 	if err != nil {
-		return conversationInfo, triple, err
+		return conversationInfo, triple, maxMessages, err
 	}
 	if len(res.Msgs) != len(messageIDs) {
-		return conversationInfo, triple, libkb.UnexpectedChatDataFromServer{Msg: fmt.Sprintf("unexpected number of messages (got %d, expected %d) from GetMessagesRemote", len(res.Msgs), len(messageIDs))}
+		return conversationInfo, triple, maxMessages, libkb.UnexpectedChatDataFromServer{Msg: fmt.Sprintf("unexpected number of messages (got %d, expected %d) from GetMessagesRemote", len(res.Msgs), len(messageIDs))}
 	}
 
 	for _, b := range res.Msgs {
 		unboxed, err := h.boxer.unboxMessage(ctx, newKeyFinder(), b)
 		if err != nil {
-			return conversationInfo, triple, err
+			return conversationInfo, triple, maxMessages, err
 		}
 
+		maxMessages = append(maxMessages, unboxed)
+
 		if len(unboxed.MessagePlaintext.MessageBodies) > 0 {
 			body := unboxed.MessagePlaintext.MessageBodies[0]
 			if t, err := body.MessageType(); err != nil {
-				return conversationInfo, triple, err
+				return conversationInfo, triple, maxMessages, err
 			} else if t == chat1.MessageType_METADATA {
 				conversationInfo.TopicName = body.Metadata().ConversationTitle
 			}
 		}
 
-		// at this point, this has to be the newest message across the
-		// conversation. Use this for TLF name
-		conversationInfo.TlfName = unboxed.MessagePlaintext.ClientHeader.TlfName
+		if unboxed.ServerHeader.MessageID.String() == conversationRemote.ReaderInfo.MaxMsgid.String() {
+			conversationInfo.TlfName = unboxed.MessagePlaintext.ClientHeader.TlfName
+		}
```

Comment:
Then it doesn't get set. One of them should be equal to `MaxMsgid`, and only that message should be used to set the TlfName. But that's a good point since otherwise it should be an error. I'll add check to error in that case